### ISSUE TITLE
/tmp/docker.sock: no such file or directory

### ISSUE THREAD

**Issue by pdiegmann** [jwilder/nginx-proxy#166]
Hello all,

I'm using docker on my Synology DS1513+, but sadly, mounting of the docker socket does not seem to be working on my system.
I appended a screenshot of my docker-run-configuration. Nothing special there, except, I use 81 as external port, but that should not make any difference, should it?
Via SSH I checked, if ```/var/run/docker.sock``` exists, and it does. But the log seems to tell a different story. Maybe some missing permissions to mount this file?

Thank you very much for your support!

```
2015-05-14 12:27:06,stdout,	[0m	[0;33;1mdockergen.1 | 	[0;31;1m2015/05/14 10:27:06 Unable to ping docker daemon: dial unix /tmp/docker.sock: no such file or directory
2015-05-14 12:26:56,stdout,	[0m	[0;33;1mdockergen.1 | 	[0;31;1m2015/05/14 10:26:56 Unable to ping docker daemon: dial unix /tmp/docker.sock: no such file or directory
2015-05-14 12:26:56,stdout,	[0;33;1mdockergen.1 | 	[0;31;1m2015/05/14 10:26:56 error listing containers: dial unix /tmp/docker.sock: no such file or directory
2015-05-14 12:26:56,stdout,	[0;37;1mforego     | 	[0mstarting dockergen.1 on port 5100
2015-05-14 12:26:56,stdout,	[0;37;1mforego     | 	[0mstarting nginx.1 on port 5000
```

![screenshot 2015-05-14 12 31 00](https://cloud.githubusercontent.com/assets/2805818/7630339/5566282e-fa35-11e4-9244-90f1eec3bec5.png)

**Comment by md5** [comment]
Where does that Docker run configuration screenshot come from? Is that from Synology's Docker package? I had seen that they announced Docker support recently since we have a DiskStation at home, but I haven't looked at it. Can you use `docker inspect` on your `nginx-proxy` container to see what it says about the mounted volumes?

**Comment by pdiegmann** [comment]
Yes, exactly. One configures docker container on the DiskStation via a UI but in this case I configured the package via the ```docker run``` command, which is then parsed into the above screenshot.

I appended the whole output from ```docker inspect nginx-proxy```
```javascript
DS1513> docker inspect nginx-proxy
[{
    "AppArmorProfile": "",
    "Args": [
        "start",
        "-r"
    ],
    "Config": {
        "AttachStderr": false,
        "AttachStdin": false,
        "AttachStdout": false,
        "Cmd": [
            "forego",
            "start",
            "-r"
        ],
        "CpuShares": 90,
        "Cpuset": "",
        "Domainname": "",
        "Entrypoint": null,
        "Env": [
            "PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin",
            "NGINX_VERSION=1.9.0-1~jessie",
            "DOCKER_GEN_VERSION=0.3.9",
            "DOCKER_HOST=unix:///tmp/docker.sock"
        ],
        "ExposedPorts": {
            "443/tcp": {},
            "80/tcp": {}
        },
        "Hostname": "nginx-proxy",
        "Image": "registry.hub.docker.com/jwilder/nginx-proxy:latest",
        "MacAddress": "",
        "Memory": 0,
        "MemorySwap": 0,
        "NetworkDisabled": false,
        "OnBuild": null,
        "OpenStdin": false,
        "PortSpecs": null,
        "StdinOnce": false,
        "Tty": false,
        "User": "",
        "Volumes": {
            "/etc/nginx/certs": {},
            "/var/cache/nginx": {}
        },
        "WorkingDir": "/app"
    },
    "Created": "2015-05-14T16:50:27.457931709Z",
    "Driver": "aufs",
    "ExecDriver": "native-0.2",
    "ExecIDs": null,
    "HostConfig": {
        "Binds": [],
        "CapAdd": [],
        "CapDrop": [],
        "ContainerIDFile": "",
        "CpuShares": 90,
        "Devices": null,
        "Dns": null,
        "DnsSearch": null,
        "Env": null,
        "ExtraHosts": null,
        "IpcMode": "",
        "Links": null,
        "LxcConf": null,
        "Memory": 0,
        "NetworkMode": "",
        "Override": true,
        "PidMode": "",
        "PortBindings": {
            "80/tcp": [
                {
                    "HostIp": "0.0.0.0",
                    "HostPort": "81"
                }
            ]
        },
        "Privileged": false,
        "PublishAllPorts": false,
        "ReadonlyRootfs": false,
        "RestartPolicy": {
            "MaximumRetryCount": 0,
            "Name": ""
        },
        "SecurityOpt": null,
        "VolumesFrom": null
    },
    "HostnamePath": "/volume1/@docker/containers/a7b829c9f05a9e1340e3f98a8c17b084a40c664154d0e234eda8b03a35533239/hostname",
    "HostsPath": "/volume1/@docker/containers/a7b829c9f05a9e1340e3f98a8c17b084a40c664154d0e234eda8b03a35533239/hosts",
    "Id": "a7b829c9f05a9e1340e3f98a8c17b084a40c664154d0e234eda8b03a35533239",
    "Image": "915ac9fa4202f43be3037c170e6f9c6c8c524dd085cb3131460f7cb3dc74bc29",
    "MountLabel": "",
    "Name": "/nginx-proxy",
    "NetworkSettings": {
        "Bridge": "docker0",
        "Gateway": "172.17.42.1",
        "GlobalIPv6Address": "",
        "GlobalIPv6PrefixLen": 0,
        "IPAddress": "172.17.0.20",
        "IPPrefixLen": 16,
        "IPv6Gateway": "",
        "LinkLocalIPv6Address": "fe80::42:acff:fe11:14",
        "LinkLocalIPv6PrefixLen": 64,
        "MacAddress": "02:42:ac:11:00:14",
        "PortMapping": null,
        "Ports": {
            "443/tcp": null,
            "80/tcp": [
                {
                    "HostIp": "0.0.0.0",
                    "HostPort": "81"
                }
            ]
        }
    },
    "Path": "forego",
    "ProcessLabel": "",
    "ResolvConfPath": "/volume1/@docker/containers/a7b829c9f05a9e1340e3f98a8c17b084a40c664154d0e234eda8b03a35533239/resolv.conf",
    "RestartCount": 0,
    "SizeRootFs": 0,
    "SizeRw": 3,
    "State": {
        "Error": "",
        "ExitCode": 0,
        "FinishedAt": "0001-01-01T00:00:00Z",
        "OOMKilled": false,
        "Paused": false,
        "Pid": 5815,
        "Restarting": false,
        "Running": true,
        "StartedAt": "2015-05-14T16:50:43.573608311Z"
    },
    "SynoFinishTime": -1,
    "SynoStatus": "running",
    "SynoUpTime": 1431622243,
    "Volumes": {
        "/etc/nginx/certs": "/volume1/@docker/vfs/dir/15fb9a2908735969626decd6970904762f2f745d8fcfb3b4ad9b27b291f25c93",
        "/var/cache/nginx": "/volume1/@docker/vfs/dir/c491b309c038e4206935827973c39eb0c97636ed1e2c5897450e980bc2628857"
    },
    "VolumesRW": {
        "/etc/nginx/certs": true,
        "/var/cache/nginx": true
    }
}
]
```

**Comment by md5** [comment]
What `docker run` command did you use? That `docker inspect` output doesn't show any evidence of `/var/run/docker.sock` being mounted to `/tmp/docker.sock`.

It might be nice to know the Docker version too, though that really shouldn't make a difference since I've been using `jwilder/nginx-proxy` since Docker 1.0 (or perhaps 0.9).

**Comment by pdiegmann** [comment]
docker's version appears to bee 1.5.0 (build 7c40f68).
I can't tell the exact command used by DSM, but for the configuration of the package I used the one supplied on the project's main page (adjusted for the port numbers):
```docker run -d -p 81:81 -v /var/run/docker.sock:/tmp/docker.sock jwilder/nginx-proxy```
This is the one I gave DSM for configuration, possibly, that's not the one used for the actual running. I'll go ahead and try it via SSH, maybe that works a little bit better

**Comment by pdiegmann** [comment]
alright, running 
```docker run -d -p 81:81 -v /var/run/docker.sock:/tmp/docker.sock --name nginx-proxy jwilder/nginx-proxy``` 
via ssh works, but following 
```docker start nginx-proxy``
via ssh or via the DSM UI produces the log:
```
2015-05-14 19:13:21,stdout,	[0m	[0;33;1mdockergen.1 | 	[0;31;1m2015/05/14 17:13:21 Unable to ping docker daemon: dial unix /tmp/docker.sock: connection refused
2015-05-14 19:13:11,stdout,	[0m	[0;33;1mdockergen.1 | 	[0;31;1m2015/05/14 17:13:11 Unable to ping docker daemon: dial unix /tmp/docker.sock: connection refused
2015-05-14 19:13:11,stdout,	[0;33;1mdockergen.1 | 	[0;31;1m2015/05/14 17:13:11 error listing containers: dial unix /tmp/docker.sock: connection refused
2015-05-14 19:13:11,stdout,	[0;37;1mforego     | 	[0mstarting dockergen.1 on port 5100
2015-05-14 19:13:11,stdout,	[0m	[0;37;1mforego     | 	[0mstarting nginx.1 on port 5000
```

**Comment by md5** [comment]
I'm curious what the Docker package for DSM is using under the hood. In particular, I wonder if they're using Python and running into this issue: https://github.com/docker/docker-py/issues/546

**Comment by md5** [comment]
We haven't updated to DSM 5.2 yet, so I'm not able to test this myself.

**Comment by pdiegmann** [comment]
sounds like it could probably be the issue. I'll just file a bug report / support request at Synology's.

Thanks for your support!

**Comment by pascalberger** [comment]
I'm running into the same problem on a Synology Diskstation. @pdiegmann Did you hear back from Synology regarding this and were able to run the container?

**Comment by pdiegmann** [comment]
@pascalberger: I did hear back, but they could not offer a working solution, though they tried to be helpful. I settled with starting it via ssh, which works most of the time and sometimes survives a reboot. But I have to say that I did not find the time to investigate this issue further.

**Comment by thomasleveil** [comment]
On my Synology (running [DSM 6.0 beta](https://www.synology.com/en-global/dsm/6.0beta)) the docker sock is at `/run/docker.sock`, make sure this isn't the case for you.

Also note that you can upgrade Docker on Synology by just copying the [latest docker binary](https://docs.docker.com/engine/installation/binaries/#get-the-linux-binary) at `/volume1/@appstore/Docker/usr/bin/docker`. I'm running Docker 1.9.1 this way.

**Comment by pascalberger** [comment]
Thanks @pdiegmann & @thomasleveil. With DSM 5.2 I've docker sock at `/run/docker.sock` and `/var/run/docker.sock`. Either works. 

Error message in the log is:

    ERROR: you need to share your Docker host socket with a volume at /tmp/docker.sock

Even though I map `/run/docker.sock` or `/var/run/docker.sock` to `/tmp/docker.sock`. But I'm not sure if it even is possible to map directories/files outside of the mapped folders, since they cannot be chosen in the UI either.

**Comment by dionysiusmarquis** [comment]
I got it up and running by symlinking `/var/run/docker.sock` to `/docker/docker.sock`

**Comment by krazyjakee** [comment]
I can't get over this on docker cloud.

![image](https://cloud.githubusercontent.com/assets/623463/20068266/4e37a372-a518-11e6-8f86-b9d663fa07c8.png)

I have mounted both ways because it's not clear. What am I missing?

**Comment by desimaniac** [comment]
@dionysiusmarquis That's not working for me.

**Comment by dionysiusmarquis** [comment]
@desimaniac I symlinked `/var/run/docker.sock` to `/volume1/docker/docker.sock`.
In your docker settings you have to mount the file without `/volume1/`.
```
volume file: /docker/docker.sock
mount point: /tmp/docker.sock
```

**Comment by desimaniac** [comment]
@dionysiusmarquis Thanks for getting back. I've created the symlink (`ln -s /var/run/docker.sock /volume1/docker/docker.sock`). Verified it was there by running the command again:  (`ln: failed to create symbolic link ‘/volume1/docker/docker.sock’: File exists`). But when I run the docker command i get "docker: Error response from daemon: stat /docker/docker.sock: no such file or directory.". Im running all this in sudo.
